tests/network-tests/src/nicaea/utils/apiWrapper.ts

```diff
@@ -0,0 +1,1208 @@
+  }
+
+  public async getOpening(id: BN): Promise<Opening> {
+    return ((await this.api.query.storageWorkingGroup.openingById<Codec[]>(id))[0] as unknown) as Opening;
+  }
+
+  public async getHiringOpening(id: BN): Promise<HiringOpening> {
+    return ((await this.api.query.hiring.openingById<Codec[]>(id))[0] as unknown) as HiringOpening;
+  }
+
+  public async getWorkers(): Promise<Worker[]> {
+    return ((await this.api.query.storageWorkingGroup.workerById<Codec[]>())[1] as unknown) as Worker[];
+  }
+
+  public async getWorkerById(id: BN): Promise<Worker> {
+    return ((await this.api.query.storageWorkingGroup.workerById<Codec[]>(id))[0] as unknown) as Worker;
+  }
+
+  public async getWorkerIdByRoleAccount(address: string): Promise<BN> {
+    const workersAndIds = await this.api.query.storageWorkingGroup.workerById<Codec[]>();
+    const workers: Worker[] = (workersAndIds[1] as unknown) as Worker[];
+    const ids: WorkerId[] = (workersAndIds[0] as unknown) as WorkerId[];
+    let index: number;
+    workers.forEach((worker, i) => {
+      if (worker.role_account_id.toString() === address) index = i;
+    });
+    return ids[index!];
+  }
+
+  public async getApplicationsIdsByRoleAccount(address: string): Promise<BN[]> {
+    const applicationsAndIds = await this.api.query.storageWorkingGroup.applicationById<Codec[]>();
+    const applications: Application[] = (applicationsAndIds[1] as unknown) as Application[];
+    const ids: ApplicationId[] = (applicationsAndIds[0] as unknown) as ApplicationId[];
+    return applications
+      .map((application, index) => (application.role_account_id.toString() === address ? ids[index] : undefined))
+      .filter(index => index !== undefined) as BN[];
+  }
+
+  public async getHiringApplicationById(id: BN): Promise<HiringApplication> {
+    return ((await this.api.query.hiring.applicationById<Codec[]>(id))[0] as unknown) as HiringApplication;
+  }
+
+  public async getApplicationById(id: BN): Promise<Application> {
+    return ((await this.api.query.storageWorkingGroup.applicationById<Codec[]>(id))[0] as unknown) as Application;
+  }
+
+  public async getActiveApplicationsIdsByRoleAccount(address: string): Promise<BN[]> {
+    const applicationsAndIds = await this.api.query.storageWorkingGroup.applicationById<Codec[]>();
+    const applications: Application[] = (applicationsAndIds[1] as unknown) as Application[];
+    const ids: ApplicationId[] = (applicationsAndIds[0] as unknown) as ApplicationId[];
+    return (
+      await Promise.all(
+        applications.map(async (application, index) => {
+          if (
+            application.role_account_id.toString() === address &&
+            (await this.getHiringApplicationById(application.application_id)).stage.type === 'Active'
+          ) {
+            return ids[index];
+          } else {
+            return undefined;
+          }
+        })
+      )
+    ).filter(index => index !== undefined) as BN[];
```

Comment:
the iteration in the find is over application ids not indices, so just the name index is incorrect, but the logic is correct.

Reply:
Done